Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.DataGridViewCell.cs

```diff
@@ -0,0 +1,1265 @@
+﻿// CodeContracts
+// 
+// Copyright (c) Microsoft Corporation
+// 
+// All rights reserved. 
+// 
+// MIT License
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System;
+using System.Diagnostics.Contracts;
+using System.ComponentModel;
+using System.Drawing;
+
+namespace System.Windows.Forms
+{
+    /// <summary>
+    /// Represents an individual cell in a <see cref="T:System.Windows.Forms.DataGridView"/> control.
+    /// </summary>
+   
+    public abstract class DataGridViewCell : DataGridViewElement //, ICloneable, IDisposable
```

Comment:
Almost all preconditions for this class are not correct.

Current implementation assumes that DataGridView property should not be null, but this is not the case.

I'm leaving this comment as a reminder for myself and for the author to double check all the preconditions for this class. I've made some proposals for some of them, but then I've realized that the same issue is happening over and over and I gave up.